cachito/workers/scm.py

```diff
@@ -138,7 +153,12 @@ def fetch_source(self):
         )
         for previous_archive in previous_archives:
             try:
-                self.update_and_archive(previous_archive)
+                # Excluding previous archives with submodules since there is
+                # no straight-forward way to get rid of the previously included
+                # submodules
+                if "-with-submodule" in str(previous_archive):
```

Comment:
Is `previous_archive` not a string? Do we need to call `str()` on it?

Reply:
Yeah, it's not. It's of type `<class 'cachito.workers.paths.SourcesDir'>`

Reply:
Ah gotcha :+1: